junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TimeoutInvocationFactory.java

```diff
@@ -43,7 +43,8 @@ <T> Invocation<T> create(ThreadMode threadMode, TimeoutInvocationParameters<T> t
 		}
 		return new SameThreadTimeoutInvocation<>(timeoutInvocationParameters.getInvocation(),
 			timeoutInvocationParameters.getTimeoutDuration(), getThreadExecutorForSameThreadInvocation(),
-			timeoutInvocationParameters.getDescriptionSupplier());
+			timeoutInvocationParameters.getDescriptionSupplier(),
+			timeoutInvocationParameters.getPreInterruptCallback());
```

Comment:
The `PreInterruptCallbackInvocation` also needs to be passed (and called) to `SeparateThreadTimeoutInvocation`, doesn't it?

Reply:
The `SeparateThreadTimeoutInvocation` does not interrupt any `Thread` inside. It justs throws a `TimeoutException` on the waiting Thread.

So I guess we should not call the `PreInterruptCallback`.

Reply:
It calls `assertTimeoutPreemptively` which calls `ExecutorService.shutdownNow()` which calls `Thread.interrupt`.

https://github.com/junit-team/junit5/blob/6593317c15fb556febbde11914fa7afe00abf8cd/junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertTimeoutPreemptively.java#L86

Reply:
It calls `assertTimeoutPreemptively` which calls `ExecutorService.shutdownNow()` which calls `Thread.interrupt`.

https://github.com/junit-team/junit5/blob/6593317c15fb556febbde11914fa7afe00abf8cd/junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertTimeoutPreemptively.java#L86